test/index.js

```diff
@@ -73,6 +89,23 @@ describe('pool', function () {
       })
     })
 
+    it('does not call setupConnection when an error occurs', function (done) {
+      var called = false
+      var pool = new Pool({
+        host: 'no-postgres-server-here.com',
```

Comment:
Maybe start a `net.Server` on a random port and use it as the `host` instead, to make the test fully local.